xcp_d/workflow/anatomical.py

```diff
@@ -269,1218 +261,932 @@ def init_anatomical_wf(
+        mem_gb=mem_gb,
+        n_procs=omp_nthreads,
+    )
+
+    # fmt:off
+    workflow.connect([
+        (inputnode, disassemble_h5_inv, [("template_to_t1w_xform", "in_file")]),
+    ])
+    # fmt:on
+
+    # convert affine from ITK binary to txt
+    convert_ants_transform = pe.Node(
+        ConvertTransformFile(dimension=3),
+        name="convert_ants_transform",
+    )
+
+    # fmt:off
+    workflow.connect([
+        (disassemble_h5, convert_ants_transform, [("affine_transform", "in_transform")]),
+    ])
+    # fmt:on
+
+    # change xfm type from "AffineTransform" to "MatrixOffsetTransformBase"
+    # since wb_command doesn't recognize "AffineTransform"
+    # (AffineTransform is a subclass of MatrixOffsetTransformBase
+    # which makes this okay to do AFAIK)
+    change_xfm_type = pe.Node(ChangeXfmType(), name="change_xfm_type")
+
+    # fmt:off
+    workflow.connect([
+        (convert_ants_transform, change_xfm_type, [("out_transform", "in_transform")]),
+    ])
+    # fmt:on
+
+    # fmt:off
+    workflow.connect([
+    ])
+    # fmt:on
+
+    # convert affine xfm to "world" so it works with -surface-apply-affine
+    convert_xfm2world = pe.Node(
+        ConvertAffine(fromwhat="itk", towhat="world"),
+        name="convert_xfm2world",
+    )
+
+    # fmt:off
+    workflow.connect([
+        (change_xfm_type, convert_xfm2world, [("out_transform", "in_file")]),
+    ])
+    # fmt:on
+
+    # merge new components
+    merge_xfms_list = pe.Node(
+        niu.Merge(2),
+        name="merge_xfms_list",
+        mem_gb=mem_gb,
+        n_procs=omp_nthreads,
+    )
+    merge_inv_xfms_list = pe.Node(
+        niu.Merge(2),
+        name="merge_inv_xfms_list",
+        mem_gb=mem_gb,
+        n_procs=omp_nthreads,
+    )
+
+    # fmt:off
+    workflow.connect([
+        (disassemble_h5, merge_xfms_list, [("displacement_field", "in1")]),
+        (disassemble_h5, merge_xfms_list, [("affine_transform", "in2")]),
+        (disassemble_h5_inv, merge_inv_xfms_list, [("displacement_field", "in2")]),
+        (disassemble_h5_inv, merge_inv_xfms_list, [("affine_transform", "in1")]),
+    ])
+    # fmt:on
+
+    # combine the affine and warpfield xfms from the
+    # disassembled h5 into a single warpfield xfm
+    combine_xfms = pe.Node(
+        ApplyTransforms(
+            reference_image=mnitemplate,
+            interpolation="LanczosWindowedSinc",
+            print_out_composite_warp_file=True,
+            output_image="ants_composite_xfm.nii.gz",
+        ),
+        name="combine_xfms",
+        mem_gb=mem_gb,
+        n_procs=omp_nthreads,
+    )
+    combine_inv_xfms = pe.Node(
+        ApplyTransforms(
+            reference_image=mnitemplate,
+            interpolation="LanczosWindowedSinc",
+            print_out_composite_warp_file=True,
+            output_image="ants_composite_inv_xfm.nii.gz",
+        ),
+        name="combine_inv_xfms",
+        mem_gb=mem_gb,
+        n_procs=omp_nthreads,
+    )

```

Comment:
@madisoth, both of these nodes take inputs, but don't feed out to anything in the workflow. Is there a reason for this?

Reply:
At one point I planned to use the combined warps on the surfaces when possible (i.e. when the surfaces fit within the FOV of the warpfield), and fall through to using the separate affine and warpfield xfms otherwise); feel free to remove this

Reply:
Thanks. I'll remove them then.